Implement multiprocessing pool

PaulNdrei opened the pull request:
Implemented multiprocessing pool on `/api/v2/tts` endpoint
Added use_mp (multiprocessing)  argument, default value `false`
Added mp_workers (number of CPUs to use, default value `mp.cpu_cores()`)

I setted use_mp (multiprocessing) value by default is false, it would better if it's true by default?

PaulNdrei committed:
Implement multiprocessing pool

gullabi commented on server/server.py:
```diff
@@ -254,6 +262,30 @@ async def details(request: Request):
+    else:
+        input_speaker_id = speaker_id
+
+    synthesizer = Synthesizer(
+        tts_checkpoint=model_path,
+        tts_config_path=config_path,
+        tts_speakers_file=speakers_file_path,
+        tts_languages_file=None,
+        vocoder_checkpoint=vocoder_path,
+        vocoder_config=vocoder_config_path,
+        encoder_checkpoint="",
+        encoder_config="",
+        use_cuda=args.use_cuda,
+    )
```
From what I see in the execution, worker initializes the model, each time it wants to do inference. This introduces overhead. In principle worker has to initialize the model only once in the beginning and in each inference step should just do inference taking advantage of the loaded model. I don't have a suggestion now, we talk next week.

gullabi replied:
@mllopartbsc can you take a look at this, this week? Thanks.

PaulNdrei committed:
Set Spawn as multiprocessing start method

PaulNdrei commented:
I added `set_start_method` for avoid the load of Synthesizer multiple times, but I guess in the end spawn method will do the same in background.

mllopartbsc committed:
Multiprocessing working commit

mllopartbsc committed:
Deleted test files

PaulNdrei committed:
Merge branch 'main' of https://github.com/projecte-aina/tts-api into multiprocessing-pool

PaulNdrei committed:
Eliminated code duplication through refactoring and optimized Helm chart and Docker for multiprocessing.

mllopartbsc committed:
Fix to memory leaking of websocket. In addition, I also changed an error occuring with websocket where the system was picking up empty sentences.

For this commit to work, it may be neede to add this changes:

if audio_time != 0:
    print(f" > Real-time factor: {process_time / audio_time}")
else:
    print(" > Audio time is zero, cannot compute real-time factor.")

To line 465 from python3.10/site-packages/TTS/utils/synthesizer.py.

PaulNdrei commented:
@mllopartbsc I reviewed your last commit, but the memory increase issue persists on websocket endpoint.

gullabi commented:
Thanks to both. Now I was revising the code and I saw that the `/api/v2/tts` endpoint is gone. For any particular reason? How does one do the Jambonz integration now?

PaulNdrei commented:
> Thanks to both. Now I was revising the code and I saw that the `/api/v2/tts` endpoint is gone. For any particular reason? How does one do the Jambonz integration now?

@gullabi 
We overwrite the default `/api/tts ` with the Jambonz-based Requestmodel. And we have removed `/api/v2/tts` .
